packages/fusion/execution/src/execution.ts

```diff
@@ -275,7 +275,15 @@ export function executeResolverOperationNodesWithDependenciesInParallel({
         mapAsyncIterator(depOpResult$ as AsyncIterableIterator<any>, handleDepOpResult),
       );
     } else if (isPromise(depOpResult$)) {
-      dependencyPromises.push(depOpResult$.then(handleDepOpResult));
+      dependencyPromises.push(
+        depOpResult$.then(res => {
+          if (isAsyncIterable(res)) {
+            return mapAsyncIterator(res as AsyncIterableIterator<any>, handleDepOpResult);
```

Comment:
Why do you have to explicitly cast it ? `isAsyncIterable` isn't already there to narrow the type ?

Reply:
Tricky, `executeResolverOperationNode` is quite complex and needs more brain power to type properly. I've left it for @ardatan to fix because he understands the intricacies of the function.